--- kphotoalbum-4.7.1/Utilities/Util.cpp.orig	2016-03-08 15:38:53.405486230 +0000
+++ kphotoalbum-4.7.1/Utilities/Util.cpp	2016-03-08 15:39:01.378406147 +0000
@@ -124,7 +124,7 @@
                 info += i18nc("aspect ratio"," (1:1)");
             else
                 info += i18nc("aspect ratio"," (1:%1)"
-                              ,QLocale::system().toString(1.0d/aspect, 'f', 2));
+                              ,QLocale::system().toString(1.0/aspect, 'f', 2));
             AddNonEmptyInfo(i18n("<b>Image Size: </b> "), info, &result);
         }
     }
